ml/base/config.py

```diff
@@ -38,6 +39,27 @@ def from_dict(cls, dic):
     def from_internal_file(cls, config_name):
         pass
 
+    @classmethod
+    def override_with_flag(cls, args_string):
+	args_list=args_string.split('--')
+	
+	#Go through all the flags.
+	for attr_arg in args_list:
+		
+		single_arg=arg.split(' ')
+		attr_name=single_arg[0].split('__')
+		attr_val=single_arg[1]
+		#For each flag, call the recursive function to populate values to the subfield.
+		populate_attr(cls,0,attr_name,attr_val)
+
+	def populate_attr(cls,i,attr_name,attr_val):
```

Comment:
in general it's bad form to use `i` in anywhere other than a loop index. Always use meaningful variable names and esp. for function args.

Reply:
Also attr_name is actually a list, this is counter-intuitive, so we should have better naming. Always name and abstract your code so that you can actually read the code aloud and never need to explain with other words, that's the goal we are aiming at.